root/account/DeleteOwnAccount.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * @flow strict
+ * Copyright (C) 2022 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import UserAccountLayout, {
+  sanitizedAccountLayoutUser,
+} from '../components/UserAccountLayout.js';
+import {CONTACT_URL} from '../constants.js';
+import {CatalystContext} from '../context.mjs';
+import FormCsrfToken
+  from '../static/scripts/edit/components/FormCsrfToken.js';
+import FormSubmit from '../static/scripts/edit/components/FormSubmit.js';
+
+type DeleteOwnAccountFormT = FormT<{
+  ...SecureConfirmFormT,
+}>;
+
+type Props = {
+  +form: DeleteOwnAccountFormT,
+};
+
+const DeleteOwnAccount = ({
+  form,
+}: Props): React$Element<typeof UserAccountLayout> => {
+  const $c = React.useContext(CatalystContext);
+  const user = $c.user;
+
+  return (
+    <UserAccountLayout
+      /* $FlowIgnore[incompatible-call] as user cannot be undefined */
```

Comment:
It shouldn’t be needed. In comparison `root/account/ChangePassword.js` just tests `user` before using it.

Reply:
That form tests `user` because it can also be used without one (for mandatory reset while logged out IIRC) so it has a valid return if `user` does not exist. This seems a bit different, but I'm not too worried about using the `$c` one - maybe it should then also be tested though rather than adding a `$FlowIgnore`.

Reply:
Fair enough. Actually I didn’t know what to return in the opposite case. I just found the solution of throwing an error instead; See <https://github.com/metabrainz/musicbrainz-server/compare/392edbe98c29d5f507c227e7fe40000ccced0fb1..66024ca025cc4b308a8b8543ee2ab9b06609acd7>.

Reply:
We also have a couple helpers you could use in the future (both throw errors): `invariant(user)` or `const user = expect($c.user)`. `invariant` will be auto-imported if you use it, but `expect` currently has to be imported manually from invariant.mjs.

Reply:
Thanks for the reminder. I changed my code for using `invariant` as it allows to return a custom message; See https://github.com/metabrainz/musicbrainz-server/compare/66024ca025cc4b308a8b8543ee2ab9b06609acd7..0ecba92e46bd5a86d0ff62ba94940b7ee8da83fa

I retested it all.